src/jvmMain/kotlin/com/natpryce/hamkrest/jvm_describe.kt

```diff
@@ -3,7 +3,7 @@ package com.natpryce.hamkrest
 import java.util.ServiceLoader
 import kotlin.reflect.KClass
 
-private val descriptionServices = ServiceLoader.load(ValueDescription::class.java)
+private val descriptionServices = synchronized(ValueDescription::class) { ServiceLoader.load(ValueDescription::class.java) }
```

Comment:
I'm going to implement this fix manually, and use a private mutex object rather than synchronizing on a public class.